test/functional/tool_wallet.py

```diff
@@ -36,8 +42,32 @@ def assert_tool_output(self, output, *args):
         assert_equal(stderr, '')
         assert_equal(stdout, output)
 
-    def run_test(self):
+    def wallet_path(self):
+        return os.path.join(self.nodes[0].datadir, 'regtest', 'wallets', 'wallet.dat')
+
+    def wallet_shasum(self):
+        h = hashlib.sha1()
+        mv = memoryview(bytearray(BUFFER_SIZE))
+        with open(self.wallet_path(), 'rb', buffering=0) as f:
+            for n in iter(lambda : f.readinto(mv), 0):
+                h.update(mv[:n])
+        return h.hexdigest()
+
+    def wallet_timestamp(self, op):
+        timestamp = os.path.getmtime(self.wallet_path())
+        self.log.info('Wallet file timestamp {0}: {1}'.format(op, timestamp))
+        return timestamp
 
+    def log_wallet_permissions(self):
+        result = oct(os.lstat(self.wallet_path()).st_mode)[-3:]
+        self.log.info('Verifying wallet file permissions: {}'.format(result))
+
+    def log_wallet_timestamp_comparison(self, old, new):
+        result = 'unchanged' if new == old else 'increased!'
+        self.log.info('Wallet file timestamp {}\n'.format(result))
```

Comment:
This looks like a test-dev debug statement. Maybe use `self.log.debug`?

Reply:
Done, switched to loglevel `debug` for all the timestamp and shasum logging.